pkgs/development/python-modules/python-watcherclient/default.nix

```
@@ -0,0 +1,75 @@
+{
+  lib,
+  buildPythonPackage,
+  cliff,
+  fetchFromGitea,
+  keystoneauth1,
+  openstackdocstheme,
+  osc-lib,
+  oslo-i18n,
+  oslo-serialization,
+  oslo-utils,
+  pbr,
+  pythonOlder,
+  setuptools,
+  sphinxcontrib-apidoc,
+  sphinxHook,
+  stestr,
+}:
+
+buildPythonPackage rec {
+  pname = "python-watcherclient";
+  version = "4.4.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitea {
+    domain = "opendev.org";
+    owner = "openstack";
+    repo = "python-watcherclient";
+    rev = version;
+    hash = "sha256-lDdiZKaeteKZEyfjpBx8KY+0FLFOYAnQXl0pTbqq0Ss=";
+  };
+
+  env.PBR_VERSION = version;
+
+  build-system = [
+    openstackdocstheme
+    pbr
+    setuptools
+    sphinxcontrib-apidoc
+    sphinxHook
+  ];
```

```suggestion
  build-system = [
    pbr
    setuptools
  ];
  
  nativeBuildInputs = [
    openstackdocstheme
    sphinxcontrib-apidoc
    sphinxHook
  ];
```
`build-system` corresponds to `build-system` in pyproject.toml, and `setup_requires` in setup.py.
https://opendev.org/openstack/python-watcherclient/src/commit/56e47c38e209010bbd9f3250ee959efaa47e61dd/setup.py#L19